1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [4]:
bonus = 5/100
year_of_service = float(input("Year Of Service: "))

if year_of_service > 5:
    user_salary = float(input("User Salary: "))
    net_bonus = user_salary * bonus
    print(f"Net Bonus Amount is {net_bonus}")
    
else:
    print("No Bonus For You")

Year Of Service: 6
User Salary: 15000
Net Bonus Amount is 750.0



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [6]:
age = int(input("Enter Your Age: "))

if age > 17:
    print("Person is Eligible for Voting")

else:
    print("Person is Not Eligible for Voting")

Enter Your Age: 18
Person is Eligible for Voting



3) Write a program to check whether a number entered by user is even or odd.

In [10]:
number = int(input("Enter a Number: "))

if number % 2 == 0:
    print("Number is Even")

else:
    print("Number is Odd")

Enter a Number: 50
Number is Even


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [13]:
number = int(input("Enter a Number: "))

if number % 7 == 0:
    print("Number is Divisible By 7")

else:
    print("Number is Not Divisible By 7")

Enter a Number: 21
Number is Divisible By 7


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [15]:
number = int(input("Enter a Number: "))

if number % 5 == 0:
    print("Hello")

else:
    print("Bye")

Enter a Number: 21
Bye


7) Write a program to display the last digit of a number.


In [17]:
number = int(input("Enter a Number: "))
last_digit_of_number = number % 10
print(f"Last Digit Of A Number '{number}': {last_digit_of_number}")

Enter a Number: 150000
Last Digit Of A Number '150000': 0


Q8. Write a program to check whether the last digit of a number( entered by user ) is 
divisible by 3 or not.

In [19]:
number = int(input("Enter a Number: "))
last_digit_of_number = number % 10

if last_digit_of_number % 3 == 0:
    print("Last Digit is Divisible by 3")

else:
    print("Last Digit is Not Divisible by 3")

Enter a Number: 150
Last Digit is Divisible by 3


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [21]:
length = int(input("Enter the Length: "))
breadth = int(input("Enter the Breadth: "))

if length == breadth:
    print("It is a Square")

else:
    print("It is a Rectangle")

Enter the Length: 7
Enter the Breadth: 7
It is a Square


10) Take two int values from user and print greatest among them.

In [22]:
num1 = int(input("Enter the First Number: "))
num2 = int(input("Enter the Second Number: "))

if num1 > num2:
    print(f"The Greatest Among Both is {num1}")

else:
    print(f"The Greatest Among Both is {num2}")

Enter the First Number: 15
Enter the Second Number: 17
The Greatest Among Both is 17


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [26]:
quantity = int(input("Enter Quantity: "))
cost_of_quantity = 100
purchase_amount = quantity * cost_of_quantity

if purchase_amount > 1000:
    discount = purchase_amount * 10/100
    total = purchase_amount - discount
    print(total)

else:
    total = purchase_amount
    print(total)

Enter Quantity: 11
990.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [27]:
marks = int(input("Enter Your Marks: "))

if marks < 25:
    print("Your Grade is F")

elif marks >= 25 and marks < 45:
    print("Your Grade is E")

elif marks >= 45 and marks < 50:
    print("Your Grade is D")

elif marks >= 50 and marks < 60:
    print("Your Grade is C")

elif marks >= 60 and marks < 80:
    print("Your Grade is B")

else:
    print("Your Grade is A")

Enter Your Marks: 48
Your Grade is D


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

Number of classes held

Number of classes attended.

And print

percentage of class attended

Is student is allowed to sit in exam or not.


In [28]:
number_of_classes_held = int(input("Enter Total Number of Classes Held: "))
number_of_classes_attended = int(input("Enter Total Number of Classes Attended: "))
percentage_of_class_attended = number_of_classes_attended / number_of_classes_held * 100

if percentage_of_class_attended < 75:
    print("Student is not Allowed to Sit in Exam")

else:
    print(f"Student is Allowed to Sit in Exam with {percentage_of_class_attended}% Attendance")

Enter Total Number of Classes Held: 18
Enter Total Number of Classes Attended: 12
Student is not Allowed to Sit in Exam


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [31]:
number_of_classes_held = int(input("Enter Total Number of Classes Held: "))
number_of_classes_attended = int(input("Enter Total Number of Classes Attended: "))
medical_cause = input("Did you have a Medical Cause ( Y or N ): ")
percentage_of_class_attended = number_of_classes_attended / number_of_classes_held * 100

if medical_cause == "Y" and percentage_of_class_attended < 75:
    print(f"Student is Allowed to Sit in Exam Due to Medical Cause with {percentage_of_class_attended}% Attendance")

elif percentage_of_class_attended < 75:
    print("Student is not Allowed to Sit in Exam")

else:
    print(f"Student is Allowed to Sit in Exam with {percentage_of_class_attended}% Attendance")

Enter Total Number of Classes Held: 60
Enter Total Number of Classes Attended: 30
Did you have a Medical Cause ( Y or N ): Y
Student is Allowed to Sit in Exam Due to Medical Cause with 50.0% Attendance


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [46]:
year = int(input("Enter Year to check Leap Year: "))

if (year % 400 == 0) or (year % 100 != 0) and (year % 4 == 0):
    print(f"Year {year} is a Leap Year")

else:
    print(f"Year {year} is not a Leap Year")

Enter Year to check Leap Year: 2052
Year 2052 is a Leap Year


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [52]:
age = int(input("Enter Your Age: "))
gender = input("Enter Your Gender ( M or F ): ")
marital_status: input("Enter Your Marital Status ( Y or N ): ")

if gender == "F":
    print("You Will Work Only in Urban Areas")

elif gender == "M" and 20 < age < 40:
    print("You Can Work Anywhere")

elif gender == "M" and 40 < age < 60:
        print("You Will Work Only in Urban Areas")

else:
    print("Error")

Enter Your Age: 25
Enter Your Gender ( M or F ): M
Enter Your Marital Status ( Y or N ): Y
You Can Work Anywhere


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

In [61]:
number_of_units = int(input("Enter the Number of Units: "))

if number_of_units <= 100:
    print(f"Electricity bill of {number_of_units} is Rs. 0")

elif number_of_units > 100 and number_of_units <= 200:
    cost_of_units_upto_200 = number_of_units * 5
    print(f"Electricity bill of {number_of_units} unit is Rs. {cost_of_units_upto_200}")

else:
    total_cost_of_units = number_of_units * 10
    print(f"Electricity bill of {number_of_units} unit is Rs. {total_cost_of_units}")

Enter the Number of Units: 150
Electricity bill of 150 unit is Rs. 750


13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [65]:
person1_age = int(input("Enter Age of Person 1: "))
person2_age = int(input("Enter Age of Person 2: "))
person3_age = int(input("Enter Age of Person 3: "))

if person1_age > person2_age and person1_age > person3_age:
    print(f"Person 1 with age {person1_age} is the Oldest Among All")

elif person2_age > person1_age and person2_age > person3_age:
    print(f"Person 2 with age {person2_age} is the Oldest Among All")

else:
    print(f"Person 3 with age {person3_age} is the Oldest Among All")


if person1_age < person2_age and person1_age < person3_age:
    print(f"Person 1 with age {person1_age} is the Youngest Among All")

elif person2_age < person1_age and person2_age < person3_age:
    print(f"Person 2 with age {person2_age} is the Youngest Among All")

else:
    print(f"Person 3 with age {person3_age} is the Youngest Among All")


Enter Age of Person 1: 2
Enter Age of Person 2: 3
Enter Age of Person 3: 1
Person 2 with age 3 is the Oldest Among All
Person 3 with age 1 is the Youngest
